File: leetcode/validParenthesis/paren.py
def isValid(s: str) -> bool:
        keys = {
            ")":"(",
            "}":"{",
            "]":"["
        }

        x = []

        for i in s:
            x.append(i)

        for _ in range(len(x)//2):
            if keys[x.pop()] == x.pop():
                continue
            else:
                return False
        return True

print(isValid(s="()")) # True

File: leetcode/validParenthesis/paren.ts
interface stringHashMap {
  [key: string]: string;
}

function isValid(s: string): boolean {
  const keys: stringHashMap = {
    ")": "(",
    "}": "{",
    "]": "[",
  };

  for (let i = 0; i < s.length; i++) {
    if (keys[s.charAt(s.length - 1)] === s.charAt(s.length - 2)) {
      continue;
    } else {
      return false;
    }
  }
  return true;
}
